smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/auth/http/integration/AddHttpAuthSchemePlugin.java

```diff
@@ -43,45 +41,76 @@
 @SmithyInternalApi
 public final class AddHttpAuthSchemePlugin implements HttpAuthTypeScriptIntegration {
     /**
-     * Integration should only be used if `experimentalIdentityAndAuth` flag is true.
+     * Integration should be skipped if the `useLegacyAuth` flag is true.
      */
     @Override
     public boolean matchesSettings(TypeScriptSettings settings) {
-        return settings.getExperimentalIdentityAndAuth();
+        return !settings.useLegacyAuth();
     }
 
     @Override
     public List<RuntimeClientPlugin> getClientPlugins() {
+        Map<String, ClientWriterConsumer> httpAuthSchemeParametersProvider = Map.of(
+            "httpAuthSchemeParametersProvider", (w, clientBodySection) -> {
```

Comment:
This is really pretty hard to follow. Consider breaking out the consumers into static methods then you can have: 
```
Map.of(
       "httpAuthSchemeParametersProvider", AddHttpAuthSchemePlugin::parametersProvider, 
       ... etc
)
```